nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -114,6 +126,83 @@ public static DocumentModel retrieveContentFromColdStorage(CoreSession session,
         return session.saveDocument(documentModel);
     }
 
+    /**
+     * Checks if the retrieved cold storage content are available for downloading.
+     *
+     * @implSpec: This method will query all documents with a cold storage content which are being retrieved
+     *            {@value COLD_STORAGE_BEING_RETRIEVED_PROPERTY},
+     *            {@link #retrieveContentFromColdStorage(CoreSession, DocumentRef, int)} and it checks if it is
```

Comment:
Maybe remove the reference to `retrieveContentFromColdStorage` and put in in a `@see`? It's not clear within the sentence.